### PR TITLE
Fix jaeger spans key names for filtering

### PR DESCRIPTION
**Pull request by joshuali925**
### Description
- make tag.error optional
- use correct field names when filtering spans

### Issues Resolved
[List any issues this PR will resolve]

### Check List
- [ ] New functionality includes testing.
  - [x] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: public/components/trace_analytics/components/traces/span_detail_flyout.tsx [hunk 1]
```diff
@@ -27,6 +27,47 @@ import { handleSpansFlyoutRequest } from '../../requests/traces_request_handler'
 import { microToMilliSec, nanoToMilliSec } from '../common/helper_functions';
 import { FlyoutListItem } from './flyout_list_item';
 
+type SpanField =
```
[the remark this answers is not in the document]
Comment (joshuali925) [reply public/components/trace_analytics/components/traces/span_detail_flyout.tsx]: thanks, updated

File: public/components/trace_analytics/requests/traces_request_handler.ts [hunk 2]
```diff
@@ -230,15 +228,35 @@ const hitsToSpanDetailData = async (hits: any, colorMap: any, mode: TraceAnalyti
   };
   if (hits.length === 0) return data;
 
-  const minStartTime = mode === 'jaeger' ? microToMilliSec(hits[hits.length - 1].sort[0]) : nanoToMilliSec(hits[hits.length - 1].sort[0]);
+  const minStartTime =
+    mode === 'jaeger'
```
Comment (mengweieric) [inline public/components/trace_analytics/requests/traces_request_handler.ts:232]: Minor: Do we want to move all these mode checking to the top like: isJaeger = mode === 'jaeger' and replace all mode checkings to isJaeger ? ...

File: public/components/trace_analytics/components/traces/flyout_list_item.tsx [hunk 3]
```diff
@@ -27,12 +27,15 @@ export function FlyoutListItem(props: FlyoutListItemProps) {
     props.description !== '-' ? (
       <EuiFlexGroup gutterSize="none">
         <EuiFlexItem>
-          <EuiText size="s" style={{ wordBreak: 'break-all', wordWrap: 'break-word', whiteSpace: 'pre-line' }}>
+          <EuiText
+            size="s"
+            style={{ wordBreak: 'break-all', wordWrap: 'break-word', whiteSpace: 'pre-line' }}
```
Comment (mengweieric) [inline public/components/trace_analytics/components/traces/flyout_list_item.tsx:32]: Minor: Do we want to move these stylings to its corresponding scss file?

**Review by mengweieric — APPROVED** [review]

**Review by Swiddis — APPROVED** [review]

**Review by anirudha — APPROVED** [review]

File: public/components/trace_analytics/requests/traces_request_handler.ts (hunk 2 above)
Comment (joshuali925) [reply public/components/trace_analytics/requests/traces_request_handler.ts:232]: good idea but i'll leave refactors out from this bug fix PR, there are a lot more `mode === 'jaeger'` checks

**Comment by joshuali925** [comment]
merging in, failure unrelated https://github.com/opensearch-project/dashboards-observability/pull/1454#issuecomment-1953283327

**Comment by wbeckler** [comment]
Is there another place where functional or integration tests are added in order to verify that these changes work and that they will continue to work in the future?

**Pull request merged**